tests/platform_helper/test_command_environment.py

```diff
@@ -1365,15 +1122,98 @@ def test_create_header_rule(self, capsys):
         )
+
+    @pytest.mark.parametrize(
+        "vpc, param_value, expected",
+        [
+            (
+                "vpc1",
+                "192.168.1.1,192.168.1.2,192.168.1.3",
+                ["192.168.1.1", "192.168.1.2", "192.168.1.3"],
+            ),
+            (
+                "vpc2",
+                " 192.168.2.1 , 192.168.2.2 , 192.168.2.3 ",
+                ["192.168.2.1", "192.168.2.2", "192.168.2.3"],
+            ),
+            (
+                None,
+                "192.168.1.1,192.168.1.2,192.168.1.3",
+                ["192.168.1.1", "192.168.1.2", "192.168.1.3"],
+            ),
+        ],
+    )
+    @mock_aws
+    def test_get_env_ips(self, vpc, param_value, expected, mock_application):
+        from dbt_platform_helper.commands.environment import get_env_ips
+
+        response = boto3.client("organizations").create_organization(FeatureSet="ALL")
+        response["Organization"]["Id"]
+        create_account_response = boto3.client("organizations").create_account(
+            Email="test-email@example.com", AccountName="test"
+        )
+        account_id = create_account_response["CreateAccountStatus"]["AccountId"]
+        mock_application.environments["development"].account_id = account_id
+        mock_application.environments["development"].sessions[account_id] = boto3.session.Session()
+        vpc = vpc if vpc else "test"
+        boto3.client("ssm").put_parameter(
+            Name=f"/{vpc}/EGRESS_IPS", Value=param_value, Type="String"
+        )
+        environment = mock_application.environments["development"]
+        result = get_env_ips(vpc, environment)
+
+        assert result == expected
+
+    @mock_aws
+    def test_get_env_ips_param_not_found(self, capsys, mock_application):
+        from dbt_platform_helper.commands.environment import get_env_ips
+
+        response = boto3.client("organizations").create_organization(FeatureSet="ALL")
+        response["Organization"]["Id"]
+        create_account_response = boto3.client("organizations").create_account(
+            Email="test-email@example.com", AccountName="test"
+        )
+        account_id = create_account_response["CreateAccountStatus"]["AccountId"]
+        mock_application.environments["development"].account_id = account_id
+        mock_application.environments["development"].sessions[account_id] = boto3.session.Session()
+        environment = mock_application.environments["development"]
+
+        with pytest.raises(click.Abort):
+            get_env_ips("vpc", environment)
+
+        captured = capsys.readouterr()
+
+        assert "No parameter found with name: /vpc/EGRESS_IPS\n" in captured.out
+
+    @patch("boto3.client")
+    def test_get_rules_tag_descriptions(self, mock_boto_client):
+        from dbt_platform_helper.commands.environment import get_rules_tag_descriptions
+
+        mock_client = Mock()
+        mock_client.describe_tags.return_value = {"TagDescriptions": ["TagDescriptions"]}
```

Comment:
Might be worth using different strings so we can be confident it didn't just use the same one twice.

Reply:
Forgive me if I'm being naive here, but should we have 21 tag descriptions?

Reply:
I think it's the number of rules that's important to replicating the bug, not the number of descriptions returned. So it's import that 21 rules means the method is called twice.

Although you're right I should make it return a different string the second time. Was being lazy

Reply:
I think it's the number of rules that's important to replicating the bug, not the number of descriptions returned. So it's important that 21 rules means the method is called twice.

Although you're right I should make it return a different string the second time. Was being lazy